tidy3d/components/lumped_element.py

```diff
@@ -272,28 +378,504 @@ def _sheet_conductance(self):
+            \\epsilon(s) = \\epsilon_\\infty - \\frac{\\Delta}{\\epsilon_0 s}
+            \\frac{a_0 + a_1 s + \\dots + a_M s^M}{b_0 + b_1 s + \\dots + b_N s^N}.
+
+        The admittance is scaled depending on the geometric properties of the lumped element by
+        the scaling factor :math:`\\Delta`. Implementation is based on the equivalent medium introduced
+        by _`[1]`.
+
+        **References**
+
+        .. [1]  J. A. Pereda, F. Alimenti, P. Mezzanotte, L. Roselli and R. Sorrentino, "A new algorithm
+                for the incorporation of arbitrary linear lumped networks into FDTD simulators," IEEE
+                Trans. Microw. Theory Tech., vol. 47, no. 6, pp. 943-949, Jun. 1999.
+    """
+
+    a: tuple[pd.NonNegativeFloat, ...] = pd.Field(
+        ...,
+        title="Numerator Coefficients",
+        description="A ``tuple`` of floats describing the coefficients of the numerator polynomial. "
+        "The length of the ``tuple`` is equal to the order of the network.",
+    )
+
+    b: tuple[pd.NonNegativeFloat, ...] = pd.Field(
+        ...,
+        title="Denominator Coefficients",
+        description="A ``tuple`` of floats describing the coefficients of the denomiator polynomial. "
+        "The length of the ``tuple`` is equal to the order of the network.",
+    )
+
+    def _to_medium(self, scaling_factor: float) -> PoleResidue:
+        """Converts the :class:`AdmittanceNetwork` model directly into a :class:`PoleResidue` model
+        with proper scaling depending on the lumped element's dimensions."""
+        a = np.array(self.a) * scaling_factor
+        b = np.array(self.b)
+        return PoleResidue.from_admittance_coeffs(a, b)
+
+    @cached_property
+    def _to_admittance_function(self) -> tuple[tuple[float, ...], tuple[float, ...]]:
+        """Converts the :class:`AdmittanceNetwork` instance into a rational expression representing the
+        admittance of the network in the Laplace domain.
+        """
+        return (self.a, self.b)
+
+
+class LinearLumpedElement(RectangularLumpedElement):
+    """Lumped element representing a network consisting of resistors, capacitors, and inductors.
+
+    Notes
+    -----
+
+        Implementation is based on the equivalent medium introduced by _`[1]`.
+
+        **References**
+
+        .. [1]  J. A. Pereda, F. Alimenti, P. Mezzanotte, L. Roselli and R. Sorrentino, "A new algorithm
+                for the incorporation of arbitrary linear lumped networks into FDTD simulators," IEEE
+                Trans. Microw. Theory Tech., vol. 47, no. 6, pp. 943-949, Jun. 1999.
+    """
+
+    network: Union[RLCNetwork, AdmittanceNetwork] = pd.Field(
+        ...,
+        title="Network",
+        description="The linear element produces an equivalent medium that emulates the "
+        "voltage-current relationship described by the ``network`` field.",
+        discriminator=TYPE_TAG_STR,
+    )
+
+    def to_structure(self, grid: Grid = None) -> Structure:
```

Comment:
I'm not sure I see exactly what you mean here. I had to split up the methods into `to_geometry`, `to_structure`, and `to_medium`, because I wanted to try and make the network definition and geometry definition be independent of each other. I also need the separate `to_geometry` call for plotting purposes.

Reply:
I see, that makes sense then.